### ISSUE TITLE
Tests do not compile with gcc 4.8.2

### ISSUE THREAD

**Issue by amerry** [danielaparker/jsoncons#28]
I tried compiling the tests for 0.95 with GCC 4.8.2 (on Ubuntu 14.04), but it didn't work.

One issue is the "typename" keyword in csv_serializer.hpp at line 62 - removing it fixes that.

The other issue is probably harder to deal with - a missing noexcept specifier in json_reader.hpp at line 51 (when implementing std::error_category). The issue is that Visual Studio versions before 2015 do no support noexcept (and neither does GCC before 4.6, I believe). So you'd need some sort of macro (like BOOST_NOEXCEPT).

I also needed to add "system" as a component to find_package(Boost) in the CMake file (it wanted it to link against the error categories - I'm not sure if there's a way to use the stdlib ones).

**Comment by danielaparker** [comment]
Thanks for reporting this. 

I created a branch 0.95.1 off 0.95 and incorporated your fixes for csv_serializer and json_reader (we have a JSONCONS_NOEXCEPT macro.) I also incorporated those fixes into master.

**Comment by amerry** [comment]
It's a little hard to follow what changes you made where, since 0.95.1 appears to have a whole bunch of stuff (including disabling allocator tests?) squashed into a single commit, but I don't think you did the noexcept fix in master: https://github.com/danielaparker/jsoncons/blob/master/src/jsoncons/parse_error_handler.hpp#L187

**Comment by danielaparker** [comment]
Incorporated some fixes for warnings addressed in later code when incorporating your fixes in 0.95.1 (the allocator test leaves memory leaks as the pool needs to be freed explicitly, which requires exposing the sizes of internal json structures), admittedly these should have been separate commits. I guess I neglected to push master, will do so later today.

**Comment by amerry** [comment]
OK. Thanks for being so prompt!

**Comment by danielaparker** [comment]
The two fixes should now be on master.

**Comment by danielaparker** [comment]
Fixed and tested on master and on releases 0.95.3 and 0.96.
